ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/question_answering/SentenceHighlightingQATranslator.java

```diff
@@ -0,0 +1,244 @@
+    private static final String KEY_CONTEXT = MLInput.CONTEXT_FIELD;
+
+    /**
+     * Determines if a prediction value indicates a relevant sentence.
+     * By default, 0 means relevant and 1 means not relevant.
+     * 
+     * @param predictionValue The prediction value from the model
+     * @return true if the prediction indicates a relevant sentence, false otherwise
+     */
+    protected boolean isRelevantPrediction(long predictionValue) {
+        return predictionValue == RELEVANT_VALUE;
+    }
+
+    @Builder.Default
+    private final SentenceSegmenter segmenter = new DefaultSentenceSegmenter();
+
+    @Builder.Default
+    private final Gson gson = new Gson();
+
+    private HuggingFaceTokenizer tokenizer;
+
+    /**
+     * Creates a new translator with default settings.
+     *
+     * @return A new SentenceHighlightingQATranslator instance
+     */
+    public static SentenceHighlightingQATranslator createDefault() {
+        return builder().build();
+    }
+
+    @Override
+    public void prepare(TranslatorContext ctx) throws IOException {
+        Path path = ctx.getModel().getModelPath();
+        tokenizer = HuggingFaceTokenizer.builder().optPadding(true).optTokenizerPath(path.resolve("tokenizer.json")).build();
+    }
+
+    @Override
+    public void setArguments(Map<String, ?> arguments) {
+        // No arguments needed for this translator
+    }
+
+    @Override
+    public NDList processInput(TranslatorContext ctx, Input input) {
+        try {
+            NDManager manager = ctx.getNDManager();
+            String question = input.getAsString(0);
+            String context = input.getAsString(1);
+
+            // Segment context into sentences for later use in output processing
+            List<Sentence> sentences = segmenter.segment(context);
+            ctx.setAttachment(KEY_SENTENCES, sentences);
+            ctx.setAttachment(KEY_QUESTION, question);
+
+            if (sentences.isEmpty()) {
+                throw new IllegalArgumentException(
+                    String.format(Locale.ROOT, "No valid sentences found in the provided context: %s", context)
+                );
+            }
+
+            // Always process the entire context
+            Encoding encodings = tokenizer.encode(question, context);
+
+            NDArray indicesArray = manager.create(encodings.getIds());
+            indicesArray.setName(INPUT_IDS);
+
+            NDArray attentionMaskArray = manager.create(encodings.getAttentionMask());
+            attentionMaskArray.setName(ATTENTION_MASK);
+
+            NDArray tokenTypeIdsArray = manager.create(encodings.getTypeIds());
+            tokenTypeIdsArray.setName(TOKEN_TYPE_IDS);
+
+            return new NDList(indicesArray, attentionMaskArray, tokenTypeIdsArray);
+        } catch (Exception e) {
+            throw new RuntimeException(String.format(Locale.ROOT, "Error processing input: %s", e.getMessage()), e);
```

Comment:
Should it be a IllegalArgument Exception?

Reply:
Changed to use IllegalArgument